flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeScheduler.java

```diff
@@ -777,7 +792,12 @@ public void runIfState(State expectedState, Runnable action, Duration delay) {
         componentMainThreadExecutor.schedule(
                 () -> {
                     if (state == expectedState) {
-                        action.run();
+                        try {
+                            action.run();
+                        } catch (Throwable t) {
+                            FatalExitExceptionHandler.INSTANCE.uncaughtException(
+                                    Thread.currentThread(), t);
```

Comment:
Let's keep in mind to pass in a proper `FatalExceptionHandler` to the `DeclarativeScheduler`. Maybe add a TODO for that.